packages/next-server/server/next-server.ts

```diff
@@ -371,11 +384,28 @@ export default class Server {
     }))
   }
 
+  private handleCompression(
+    req: IncomingMessage,
+    res: ServerResponse
+  ): Promise<void> {
+    return new Promise((resolve, reject) => {
+      if (!this.compression) {
+        return resolve()
+      }
+      this.compression(req, res, err => {
+        if (err) reject(err)
+        else resolve()
+      })
+    })
+  }
+
   private async run(
     req: IncomingMessage,
     res: ServerResponse,
     parsedUrl: UrlWithParsedQuery
   ) {
+    await this.handleCompression(req, res)
```

Comment:
Should this be an `if` clause instead of an `await`? Are there any speed differences?

Reply:
Maybe just the following since we don't need `async`:
```suggestion
    if (this.compression) {
      this.compression(req, res)
    }
```

Reply:
Nice catch on passing the noop function.